docs/api.md

```diff
@@ -199,14 +204,151 @@ Once booted, we can shut a VM down from the REST API:
 curl --unix-socket /tmp/cloud-hypervisor.sock -i -X PUT 'http://localhost/api/v1/vm.shutdown'
 ```
 
+### D-Bus API
+
+Cloud Hypervisor offers a D-Bus API as an alternative to its REST API. As of
+writing this document, the D-Bus API mirrors the functionality of the REST
+API and shares the same set of endpoints, meaning that it supports every call
+that is supported by the REST API and can be a drop-in replacement since it
+also consumes/produces JSON.
+
+#### D-Bus API Location and availability
+
+This feature is not compiled into Cloud Hypervisor by default. Users who
+wish to use the D-Bus API, must explicitly enable it with the `dbus_api`
+feature flag when compiling Cloud Hypervisor.
+
+```sh
+$ ./scripts/dev_cli.sh build --release --libc musl -- --features dbus_api
+```
+
+Once this feature is enabled, it can be configured with the following
+CLI options:
+
+```
+  --dbus-service-name
+                    well known name of the service
+  --dbus-object-path
+                    object path to serve the dbus interface
+  --dbus-system-bus use the system bus instead of a session bus
+```
+
+Example invocation:
+
+```sh
+$ ./cloud-hypervisor --dbus-service-name "org.cloudhypervisor.DBusApi" \
+                     --dbus-object-path "/org/cloudhypervisor/DBusApi"
+```
+
+This will start serving a service with the name `org.cloudhypervisor.DBusApi1`
+which in turn can be used to control and manage Cloud Hypervisor.
+
+#### D-Bus API Interface
+
+Please refer to the [Cloud Hypervisor API](https://raw.githubusercontent.com/cloud-hypervisor/cloud-hypervisor/master/vmm/src/api/openapi/cloud-hypervisor.yaml)
+document for more details about the API payloads and responses. As previously
```

Comment:
Also, I think it is clearer if we add this sentence as a part of note to the REST API, say line 120 in this file. Here for the dbus section, we can simply say the dbus-api mirrors REST API. Also, I think the xml list does not provide much additional information and can be omitted. WDYT?

Reply:
> Also, I think the xml list does not provide much additional information and can be omitted. WDYT?

I believe it's a good idea to exclude the XML section from the documentation until we implement additional features into the D-Bus API.

> I think it is clearer if we add this sentence as a part of note to the REST API, say line 120 in this file. 

That sentence is already a part of the #rest-api section (lines 48-49). I believe we can relocate (I just modified according to the previous comment for the time being) it to line 120 since it makes sense to mention it there, as you also noted.

Reply:
> That sentence is already a part of the #rest-api section (lines 48-49). I believe we can relocate (I just modified according to the previous comment for the time being) it to line 120 since it makes sense to mention it there, as you also noted.

Oh I didn't see that. I think it is fine to leave it as it is, so that it is being called-out earlier in a separate section (rather than a foot note).